### PR TITLE
Added code for handling single endpoint in generate

### PR DESCRIPTION
**Pull request by SantoshNC68**

### CODE REVIEW

File: src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+public class RemremGenerateController {
+	
+    @Autowired
+	private List<MsgService> msgServices;
+	private JsonParser parser = new JsonParser();
+	
+	@RequestMapping(value="/{mp}", method = RequestMethod.POST)
+	public JsonElement generate(@PathVariable String mp,@RequestParam("msgType") String msgType,
+           @RequestBody JsonObject bodyJson) {
+		MsgService msgService = getMessageService(mp);
```
Comment (p-backman-ericsson) [inline src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java]: You have a mix of tabs and spaces in this file. Please only use spaces to indent.

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -15,8 +15,8 @@ org:
       logging:
         LoggingSystem: none
 jar:
-  path:
```
Comment (p-backman-ericsson) [inline src/main/resources/application.yml]: It is only whites pace changes in this file. It should not be part of the pull request.

File: src/test/java/com/ericsson/eiffel/remrem/generate/cli/CliUnitTests.java [hunk 3]
```diff
@@ -63,7 +65,7 @@ public void testHandleFileArgsFail() throws Exception {
         assertTrue(CLIOptions.getErrorCodes().contains(code));		
     }
 
-    @Test
+    /*@Test
```
Comment (p-backman-ericsson) [inline src/test/java/com/ericsson/eiffel/remrem/generate/cli/CliUnitTests.java]: Is it possible to enable this test case or describe why it has been commented out?

File: src/test/java/com/ericsson/eiffel/remrem/generate/service/EiffelRemremControllerUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    @Mock
+    MsgService service;
+    
+    @Mock
+    MsgService service2;
+    
+
+    @Spy
+    private List<MsgService> msgServices = new ArrayList<MsgService>();
+    static String testInputName = "samplesemantics.json";
```
Comment (p-backman-ericsson) [inline src/test/java/com/ericsson/eiffel/remrem/generate/service/EiffelRemremControllerUnitTest.java]: Since we only want to test the logic of RemremGenerateController we do not need a file with input. The input will not be used anyway since we mock the service. The body below could be a very simple JsonElement. Perhaps we could use a mocked body since we do not care about its content.

File: src/test/java/com/ericsson/eiffel/remrem/generate/service/EiffelRemremControllerUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+            .useDelimiter("\\A").next());
+        MockitoAnnotations.initMocks(this);
+        msgServices.add(service);
+        msgServices.add(service2);
+        Mockito.when(service.getServiceName()).thenReturn("eiffelsemantics");
+        Mockito.when(service2.getServiceName()).thenReturn("eiffel3");
+        
+        Mockito.when(service.generateMsg(
+            Mockito.eq("eiffelactivityfinished"),
+            Mockito.anyObject()
```
Comment (p-backman-ericsson) [inline src/test/java/com/ericsson/eiffel/remrem/generate/service/EiffelRemremControllerUnitTest.java]: Replace Mockito.anyObject with body. This validates that the body is sent to the service. 4 occurrences.

File: src/test/java/com/ericsson/eiffel/remrem/generate/service/EiffelRemremControllerUnitTest.java [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+        JsonElement elem = unit.generate("eiffel3", "eiffeljobfinished", body.getAsJsonObject());
+        assertEquals(elem.getAsJsonObject().get("result").getAsString(), "SUCCESS");
+    }
+    
+    @Test
+    public void testEiffel3FailureEvent() throws Exception{
+        JsonElement elem = unit.generate("eiffel3", "eiffelactivityfinished", body.getAsJsonObject());
+        assertEquals(elem.getAsJsonObject().get("result").getAsString(), "FAILURE");
+    }   
+}
```
Comment (p-backman-ericsson) [inline src/test/java/com/ericsson/eiffel/remrem/generate/service/EiffelRemremControllerUnitTest.java]: Would be good with a unit test using a service that do not exist. unit.generate("non_existing_protocol",...

File: src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+@RestController @RequestMapping("/*") 
+public class RemremGenerateController {
+	
+    @Autowired
+	private List<MsgService> msgServices;
+	private JsonParser parser = new JsonParser();
+	
+	@RequestMapping(value="/{mp}", method = RequestMethod.POST)
+	public JsonElement generate(@PathVariable String mp,@RequestParam("msgType") String msgType,
```
Comment (l-ohlen-ericsson) [inline src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java]: A REST method should return with an appropriate HTTP response code

File: src/test/java/com/ericsson/eiffel/remrem/generate/cli/CliUnitTests.java (hunk 3 above)
Comment (evasiba-ericsson) [inline src/test/java/com/ericsson/eiffel/remrem/generate/cli/CliUnitTests.java]: This test case should be on.

File: src/main/resources/application.yml (hunk 2 above)
Comment (SantoshNC68) [inline src/main/resources/application.yml]: I guess this is not required now as there is no code handling done by us, Will remove this change
